src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,122 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int INITIAL_ARRAY_LENGTH = 10;
+    private static final int INITIAL_ARRAY_SIZE = 0;
+    private Object[] arrayList;
+    private int size;
+
+    public ArrayList() {
+        arrayList = new Object[INITIAL_ARRAY_LENGTH];
+        size = INITIAL_ARRAY_SIZE;
+    }
+
     @Override
     public void add(T value) {
-
+        sizeUp();
+        arrayList[size - 1] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        sizeUp();
+        checkIndex(index);
+        arrayShift(index, +1);
+        arrayList[index] = value;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            sizeUp();
+            arrayList[size - 1] = list.get(i);
```

Comment:
```suggestion
            add(list.get(i));
```
можна одразу використати готовий метод